### DISCUSSION TITLE
If/Then activity done output.

### DISCUSSION THREAD

**Discussion by tomy2105** [elsa-workflows/elsa-core#788] [Q&A] [answered]
Could you please explain what does Done output on If/Else activity do? When will the activity exit on it?

**Comment by sfmskywalker** [comment, marked as answer]
Very good question.

In Elsa 2, the **Done** outcome can be used whenever you want to execute activities *after* either the **True** or **False** branch have completely executed.

This is different from Elsa 1, where the **Done** outcome is scheduled **at the same time** as either the **True** or **False** outcome.

The reason for this change is that it is now more intuitive to keep on chaining activities to the `If` activity.

**Reply by tomy2105** [reply]
Thanks for the speedy answer.

So you are saying that these two workflows are identical in functionality (one has two if/then branches "collapsed" back to one and other uses done output)? 
Or is there any difference I don't see? 
If no difference what is the use case for one and for the other?


![Horizontal](https://user-images.githubusercontent.com/7658839/112292720-acab8300-8c91-11eb-9050-2cad0d7e718f.png)
![Vertical](https://user-images.githubusercontent.com/7658839/112292724-ad441980-8c91-11eb-9568-c3372a034274.png)

**Reply by sfmskywalker** [reply]
Exactly. The Done branch will always execute after True or False has executed. I imagine that in most typical scenarios, you will only use the True and False branch, but if you need to perform logic after either one of those are finished, the Done outcome is there.

**Comment by jdevillard** [comment]
Sorry to ask on this answered question, but regarding the designer and UI point of view , for the end user, it wouldn't be better to use something like a Join with WaitAny and not another branch ? (but maybe regarding code, using this will not work.)

**Reply by sfmskywalker** [reply]
It’s a good question. And I agree that from the designer point of view, using a Fork/Join combination probably models the flow better. This is still possible of course. Perhaps it would be better to simply not display the Done outcome on the If activity. But at the same time, others might find it useful to have.
